### ISSUE TITLE
followup 0210's : languages and locales

### ISSUE THREAD

**Issue by Thorin-Oakenpants** [ghacksuserjs/ghacks-user.js#812]
questions
* [ ] something about testing non en-US packs. Someone will need to tell me what I am testing - *cough* earthlng *cough*
* [ ] `intl.locale.privacy.web_exposed` if anyone wants to follow it further
* [x] `intl.regional_prefs.use_os_locales` - is this geo (0200s) or lang/locale (0210s)
   * [bugzillas](https://bugzilla.mozilla.org/buglist.cgi?bug_id=1379420,1364789)
   * its lang/locale for sure
* [x] `intl.regional_prefs.use_os_locales` - does this have anything to do with web content?
* [x] how do you set your app language now there is no UI (the UI was removed in FF70)?
* [ ] do we try and fit `privacy.spoof_english` into this section

Edit: assigned earthlng for kicks 😀

**Comment by Thorin-Oakenpants** [comment]
OK, now it makes more sense. "Regional preferences" = formats for dates, time etc. So it's not a geo "regional" thing like regional searches. So it belongs in 0210s. Cool.

```
/* 0213: use application locale over OS locale in regional preferences [FF56+]
```
"application language" - to me that means the language Firefox uses for menus, etc. So does this even have any bearing on web-content?

**Comment by Thorin-Oakenpants** [comment]
`intl.regional_prefs.use_os_locales` is not used in TB... it's as I suspected, it's for the "application language", so we can remove it

**Comment by Thorin-Oakenpants** [comment]
`intl.regional_prefs.use_os_locales` is default false in FF and TB, so I think we can remove it.

**Comment by rusty-snake** [comment]
>  how do you set your app language now there is no UI (the UI was removed in FF70)?

The UI is behind `intl.multilingual.enabled` which is `true` in my FF69, but `false` in my FF71.

**Comment by Thorin-Oakenpants** [comment]
Ahh, that's cool. Thanks.

defaults
- 62 = does not exist or hidden
- 63 = false
- 64 to 69 = true
- 70dev/71nightly = false

So in 62/63 they rolled it out, in 64+ they flipped it on. Dev/Nightly channels for some reason it's off (or it's planned to be off in the future and is riding the trains?).

Note: it's not picked up in @earthlng 's diff (assuming it was still false in the dev version he last did the diff on) - I'll let him decide if something needs a tweak

**Comment by Thorin-Oakenpants** [comment]
Since lang/locale is now a pithy 3 prefs, I think we can easily add in spoof_english (inactive) which would tie it all up nicely

**Comment by earthlng** [comment]
> The UI is behind `intl.multilingual.enabled` which is `true` in my FF69, but `false` in my FF71.

`intl.multilingual.enabled` is `false` in Nightly because apparently [AMO only serves language packs for release and beta versions](https://dxr.mozilla.org/mozilla-central/rev/e1a65223d498aa0b8e3e4802d8267db2768073d9/browser/app/profile/firefox.js#1897-1899)

**Comment by Thorin-Oakenpants** [comment]
When ~~we~~  I mean, when I add spoof_english, I want to make sure exactly what it does. I know it sets the `0210` (accept language) to `en-US, en` but I'm not sure what it does with `0211`

**Comment by rusty-snake** [comment]
@Thorin-Oakenpants new/empty firefox profile; firefox 69; no user.js

Open `about:config` and search for `/privacy.spoof_english|javascript.use_us_english_locale/`.

RFP off:
 * `privacy.spoof_english` =  0
 * `javascript.use_us_english_locale` hidden
 *  `privacy.spoof_english` =  1
 * `javascript.use_us_english_locale` hidden
 * `privacy.spoof_english` =  2
 * `javascript.use_us_english_locale` hidden

RFP on:
 * `privacy.spoof_english` =  0
 * `javascript.use_us_english_locale` hidden
 * `privacy.spoof_english` =  1
 * `javascript.use_us_english_locale` hidden
 * `privacy.spoof_english` =  2
 * `javascript.use_us_english_locale` = true

If RFP is on and `privacy.spoof_english` gets 2, `javascript.use_us_english_locale` will be created.

1. `privacy.resistFingerprinting = true`
2. `privacy.spoof_english =  2`
3. `javascript.use_us_english_locale = true`

**Comment by Thorin-Oakenpants** [comment]
Cheers. I meant to check what it does when you use an en-US build, but you're on a different OS locale (I won't know until I'm sure, but this may be a hole). And of course what any build does when it's set **manually** vs when **prompted**. And what gets changed to if you change the value (reset it), etc.

**Comment by Thorin-Oakenpants** [comment]
> **UPDATE**: `privacy.spoof_english` doesn't do anything if set in user.js. Adding it to the user.js is a bad idea.

As already said, it'd be added inactive, so I can add the relevant info for end-users: so they know what it does, how to reset it, etc. Already had two or three issues on how to do this.

**Comment by Thorin-Oakenpants** [comment]
I'm adding a hash function to TZP's time/date section so I can easily determine if anything foes amiss. 

For the record
- manually changing `spoof` auto-magically adds/removes `use_us_english_locale` (I'll double check if this also happens with setting it from the user.js)
- toggling RFP off resets `spoof` and `use_us_english_locale`
   * there is also something else about toggling RFP on/off that affects this, and I'll have to look it up again: It depends on whether you started with RFP on or off, some things are not covered
- toggling RFP back on I haven't tested if this then prompts for spoofing again on the next HTTP* visit (on a non en-US bundle)

**Comment by Thorin-Oakenpants** [comment]
And none of that happens unless RFP = true

**Comment by Thorin-Oakenpants** [comment]
Looking loads secksier and easier to diagnose changes
 - https://ghacksuserjs.github.io/TorZillaPrint/TorZillaPrint.html#language

![seckie](https://user-images.githubusercontent.com/16656956/67137700-f1272980-f228-11e9-8970-c021a8dc1893.png)

**Comment by Thorin-Oakenpants** [comment]
Does anyone want me to improve this section. If no one speaks up, then I'll close this

**Comment by KOLANICH** [comment]
>intl.locale.privacy.web_exposed

available via [`GetWebExposedLocales`](https://dxr.mozilla.org/mozilla-central/source/dom/base/nsGlobalWindowInner.cpp#7087), but it is not called anywhere.

**Comment by Thorin-Oakenpants** [comment]
Thanks @KOLANICH .. ~~we~~ I did a search after earthlng brought it up, before I made this issue. I don't think its a problem going forward. Any changes with date/locales etc is pretty closely watched by the Tor Uplift team - and the `intl.` changes are coming thick and fast: next up (I think), first day of the week will be exposed, as well as other calendar info: but everything gets tied back to the locale exposed by RFP spoof english (or our three prefs in the user.js).

I guess I could try to find out more about it
